File: src/front/srcFront/services/perfil.ts
import { solicitarApi } from "./api";
import type { UsuarioAutenticado } from "./autenticacion";


export interface ProgresoPerfil {
  total_letras: number;
  letras_iniciadas: number;
  letras_dominadas: number;
  cantidad_intentos: number;
  cantidad_aciertos: number;
  porcentaje_progreso: number;
}

export interface PerfilUsuario extends UsuarioAutenticado {
  progreso: ProgresoPerfil;
}


export function consultarPerfil(
  token: string
): Promise<PerfilUsuario> {
  return solicitarApi<PerfilUsuario>("/perfil", {
    headers: {
      Authorization: `Bearer ${token}`
    }
  });
}

File: src/schemas/perfil.py
from pydantic import BaseModel, Field

from src.schemas.autenticacion import UsuarioAutenticadoRespuesta


class ProgresoPerfilRespuesta(BaseModel):
    total_letras: int = Field(ge=0)
    letras_iniciadas: int = Field(ge=0)
    letras_dominadas: int = Field(ge=0)
    cantidad_intentos: int = Field(ge=0)
    cantidad_aciertos: int = Field(ge=0)
    porcentaje_progreso: float = Field(ge=0, le=100)


class PerfilRespuesta(UsuarioAutenticadoRespuesta):
    progreso: ProgresoPerfilRespuesta